tgui/packages/tgui/interfaces/Vote.js

```diff
@@ -0,0 +1,229 @@
+                  disabled={!upper_admin || !avr}
+                  onClick={() => act("restart")} >
+                  Restart
+                </Button>
+                {!!upper_admin && (
+                  <Button.Checkbox
+                    ml={1}
+                    color="red"
+                    checked={!avr}
+                    onClick={() => act("toggle_restart")} >
+                    Disable{!avr ? "d" : ""}
+                  </Button.Checkbox>
+                )}
+              </Box>
+              <Box mb={1}>
+                <Button
+                  disabled={!upper_admin || !avm}
+                  onClick={() => act("gamemode")} >
+                  Gamemode
+                </Button>
+                {!!upper_admin && (
+                  <Button.Checkbox
+                    ml={1}
+                    color="red"
+                    checked={!avm}
+                    onClick={() => act("toggle_gamemode")} >
+                    Disable{!avm ? "d" : ""}
+                  </Button.Checkbox>
+                )}
+              </Box>
+            </Flex.Item>
+            <Flex.Item>
+              <Button disabled={!upper_admin} onClick={() => act("custom")}>
+                Create Custom Vote
+              </Button>
+            </Flex.Item>
+          </Flex>
+        </Collapsible>
+        <Collapsible title="View Voters">
+          <Box mt={2} width="100%" height={6} overflowY="scroll">
+            {voting.map(voter => {
+              return <Box key={voter}>{voter}</Box>;
+            })}
+          </Box>
+        </Collapsible>
+      </Section>
+    </Flex.Item>
+  );
+};
+
+// Display choices as buttons
+const ChoicesPanel = (props, context) => {
+  const { data } = useBackend(context);
+  const { mode, choices } = data;
+
+  let content;
+  if (choices.length === 0) {
+    content = "No choices available!";
+  }
+  // Single box for most normal vote types
+  else if ((choices.length < 10) | (mode === "custom")) {
+    content = (
+      <DisplayChoices
+        choices={choices}
+        tally="Votes:"
+        startIndex={0}
+        margin={1} />
+    );
+  } else {
+    // If there's both too much content, most likely gamemode
+    content = (
+      <Flex justify="space-between" direction="row">
+        <Flex direction="column">
+          <DisplayChoices
+            choices={choices.filter(
+              (choice, index) => index < choices.length / 2
+            )}
+            tally="|"
+            startIndex={0}
+            margin={0} />
+        </Flex>
+        <Flex direction="column" ml={1}>
+          <DisplayChoices
+            choices={choices.filter(
+              (choice, index) => index > choices.length / 2
+            )}
+            tally="|"
+            startIndex={Math.ceil(choices.length / 2)}
+            margin={0} />
+        </Flex>
+      </Flex>
+    );
+  }
+
+  return (
+    <Flex.Item mb={1} grow={1}>
+      <Section fill title="Choices">
```

Comment:
I feel like this may want to be scrollable just with how this voting /could/ get longer but maybe I underestimate how DisplayChoices spaces things out. if you have faith unlike me you can just ignore this suggestion

Reply:
The only thing currently pushing this is game mode, I guess it would be future proof if it were made scrollable